examples/train_decoder_only_base2.py

```diff
@@ -0,0 +1,107 @@
+from decoder_only_model import DecoderOnlyConfig, DecoderOnlyModel
+
+
+import time
+import itertools
+
+import torch
+import torch_xla.utils.utils as xu
+import torch.optim as optim
+import torch.nn as nn
+import torch_xla2
+import torch_xla2.interop
+
+
+USE_FSDPv2 = False
+
+class OptimizerStep:
+  
+  def __init__(self, model, optimizer, loss_fn):
+
+    self.model = model
+    self.optimizer = optimizer
+    self.loss_fn = loss_fn
+
+    self.state = self.optimizer.state_dict()
+    # if USE_FSDPv2:
+    #     self.model = FSDPv2(model)
+    #     self.state = FSDPv2.shard_state(self.state)
+
+    def single_step(data, target, state):
+        self.optimizer.load_state_dict(state)
+        self.optimizer.zero_grad()
+        res = torch.func.functional_call(self.model, state, data)
+        loss = self.loss_fn(res, target)
+        loss.backward()
+        self.optimizer.step()
+        return loss, self.optimizer.state_dict()
+
+    self.single_step = torch_xla2.interop.jax_jit(single_step)
+
+  def run_optimizer(self, data, target):
+    loss, self.state = self.single_step(data, target, self.state)
```

Comment:
I understand why we need a `self.state` now... but I still felt like it is confusing to normal user.